huaweicloud/config/config.go

```diff
@@ -343,6 +350,99 @@ func buildClientByAKSK(c *Config) error {
 	return genClients(c, pao, dao)
 }
 
+func buildClientByConfig(c *Config) error {
+	profilePath, err := homedir.Expand(c.SharedConfigFile)
+	if err != nil {
+		return err
+	}
+
+	current := c.Profile
+	providerConfig := make(map[string]interface{})
+	_, err = os.Stat(profilePath)
+	if !os.IsNotExist(err) {
+		data, err := ioutil.ReadFile(profilePath)
+		if err != nil {
+			return fmt.Errorf("Err reading from shared config file: %s", err)
+		}
+		sharedConfig := map[string]interface{}{}
+		err = json.Unmarshal(data, &sharedConfig)
+		if err != nil {
+			return err
+		}
+
+		// fetch current from shared config if not specified with provider
+		if current == "" {
+			if v, ok := sharedConfig["current"]; ok {
+				current = v.(string)
+			} else {
+				return fmt.Errorf("Error parsing current from shared config file")
+			}
+		}
+
+		// fetch the current profile config
+		found := false
+		if _, ok := sharedConfig["profiles"]; !ok {
+			return fmt.Errorf("Error parsing profiles from shared config file")
+		}
+		for _, v := range sharedConfig["profiles"].([]interface{}) {
+			if _, ok := v.(map[string]interface{})["name"]; !ok {
+				return fmt.Errorf("Error parsing name from shared config file profile")
+			}
+			if current == v.(map[string]interface{})["name"] {
+				providerConfig = v.(map[string]interface{})
+				found = true
+			}
+		}
+		if !found {
+			return fmt.Errorf("Error finding profile %s from shared config file", current)
+		}
+	} else {
+		return fmt.Errorf("The specified shared config file %s does not exist", profilePath)
+	}
+
+	var mode string
+	if v, ok := providerConfig["mode"]; ok {
+		mode = v.(string)
+	} else {
+		return fmt.Errorf("Error parsing mode from shared config file")
+	}
+
+	if mode == "AKSK" {
+		if v, ok := providerConfig["accessKeyId"]; ok {
+			c.AccessKey = v.(string)
+		} else {
+			return fmt.Errorf("Error parsing accessKeyId from shared config file")
+		}
+		if v, ok := providerConfig["secretAccessKey"]; ok {
+			c.SecretKey = v.(string)
+		} else {
+			return fmt.Errorf("Error parsing secretAccessKey from shared config file")
+		}
+		if c.Region == "" {
+			if v, ok := providerConfig["region"]; ok {
```

Comment:
that's because these files are not requred in the config file and defaults to empty string. This makes it possible to specify region in the provider config if it's not configured in the shared config file.